### DISCUSSION TITLE
Is it possible to link multiple field to same field?

### DISCUSSION THREAD

**Discussion by H01001000** [prisma/prisma#22925] [Q&A]
### Question

Sorry, I still grasping the idea of Prisma. 
For the following schema (illustration only), is it possible to link `orginalImage`, `highResImage`, `previewImage` in post model to Image model`post` field? So I can find all images in the database and get the post, and choose the image type from post.
```
Model Post {
  ...
  orginalImage       Image
  highResImage       Image
  previewImage       Image
}

Model Image {
  ...
  post               Post
}
```

**Comment by ludralph** [comment]
Hi @H01001000 👋 

Thank you for raising this question.

Yes, it is possible to link multiple fields to the same field in Prisma. In your case, you can link `originalImage`, `highResImage`, and `previewImage` in the `Post` model to the `post` field in the `Image` model. However, you need to define the relations explicitly in your Prisma schema. For example:
```prisma
model Post {
  id              Int    @id @default(autoincrement())
  originalImageId Int   @unique
  originalImage   Image @relation("OriginalImage", fields: [originalImageId], references: [id])
  
  highResImageId  Int @unique
  highResImage    Image @relation("HighResImage", fields: [highResImageId], references: [id])
  
  previewImageId  Int @unique
  previewImage    Image @relation("PreviewImage", fields: [previewImageId], references: [id])
}

model Image {
  id              Int    @id @default(autoincrement())
  // ... other fields
  originalPost    Post?  @relation("OriginalImage")
  highResPost     Post?  @relation("HighResImage")
  previewPost     Post?  @relation("PreviewImage")
}
```

In this schema, each image type in the Post model has a corresponding relation in the Image model. This allows you to find all images in the database, get the post, and choose the image type from the post.

The relationships in the provided schema are examples of one-to-one relationships. In Prisma, a one-to-one relationship is represented as a relation between two models where each instance of the first model relates to exactly one instance of the second model, and vice versa.
In your schema:

- Each Post can have one originalImage, one highResImage, and one previewImage.
- Each Image can be an originalImage, highResImage, or previewImage for exactly one Post.

For more information about one-to-one relationships in Prisma ORM, you can refer to the [documentation](https://www.prisma.io/docs/orm/prisma-schema/data-model/relations/one-to-one-relations).

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by KessyGuru** [reply]
Thank you this saved me a lot.

**Comment by H01001000** [comment]
Sorry I think you misunderstood the question. I know I can make it connect to post with additional field for each relationship, but I wonder if it possible to make each image have one post using the single post field. Where for all image I can use the post field to access the post. Each post can link to 3 image

**Comment by H01001000** [comment]
like 
```
SELECT *
  FROM post
  FULL OUTER JOIN image originalImage
    ON imageRequest.originalImageId = originalImage.id LEFT OUTER JOIN
	image highResImage
    ON imageRequest.highResImageId= highResImage.id LEFT OUTER JOIN
	image previewImage
    ON imageRequest.previewImageId = previewImage.id;

```

**Comment by H01001000** [comment]
@ludralph

**Reply by ludralph** [reply]
Hi @H01001000 

Would this schema work for you?
```prisma

model Post {
  id              String @id @default(cuid())
  originalImage   Image? @relation("OriginalImage")
  highResImage    Image? @relation("HighResImage")
  previewImage    Image? @relation("PreviewImage")
}

model Image {
  id              String @id @default(cuid())
  postAsOriginal  Post?  @relation("OriginalImage", fields: [postAsOriginalId], references: [id])
  postAsOriginalId String? @unique
  postAsHighRes   Post?  @relation("HighResImage", fields: [postAsHighResId], references: [id])
  postAsHighResId String? @unique
  postAsPreview   Post?  @relation("PreviewImage", fields: [postAsPreviewId], references: [id])
  postAsPreviewId String? @unique
}
```
This schema sets up a one-to-one relationship between `Post` and `Image` for each image type. Each Image can be associated with a `Post` as either an original, high-res, or preview image. The `Image` model has separate fields for each relationship type, allowing you to query in both directions.

An image can only be associated with one post at a time, but it can serve as different image types for different posts.
The schema doesn't enforce that an image can only be one type for a given post, so you'd need to handle that in your application logic. To query all images and their associated posts, you could perform a query like this:
```js
const imagesWithPosts = await prisma.image.findMany({
  include: {
    postAsOriginal: true,
    postAsHighRes: true,
    postAsPreview: true
  }
  ```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
